src/mcts/search.cc

```diff
@@ -718,17 +726,32 @@ std::pair<Node*, bool> Search::PickNodeToExtend(Node* node,
   }
 }
 
-std::pair<Move, Move> Search::GetBestMove() const {
+std::tuple<Move, Move, bool> Search::GetBestMove() const {
   SharedMutex::SharedLock lock(nodes_mutex_);
   Mutex::Lock counters_lock(counters_mutex_);
-  return GetBestMoveInternal();
-}
 
-std::pair<Move, Move> Search::GetBestMoveInternal() const
-    REQUIRES_SHARED(nodes_mutex_) REQUIRES_SHARED(counters_mutex_) {
-  if (responded_bestmove_) return best_move_;
-  if (!root_node_->HasChildren()) return {};
+  auto best = GetBestMoveInternal();
+
+  // guard for the case where we aren't ever resigning
+  if (kResignPercentage == 0) { 
+    return std::make_tuple(best.first, best.second, 0);
+  }
 
+  // if resigning is enabled, we check if we should resign.
+  // std::min()'d to make the variable name accurate 
+  // someone, *please* suggest a better name.
+  auto win_percent = std::min(
+    (GetBestNodeInternal()->GetQ(0, 0) + 1) * 100, 100.0f);
```

Comment:
I don't think this is what we want.
There are 3 options that seem valid to me.
1) An overload of GetBestNodeInternal which disables temperature to ensure we use the actual best node.
2) Use the Q of the current head instead (if that is being updated by the search code, I've not checked).
3) Use the node which corresponds to the selected move, rather than resampling again at random.
The scripts which are used to decide on a resignpct which is acceptable have been tuned on one of these three, so it should probably match those scripts - or the scripts will need updating and a new resignpct threshold decided.

I don't really like option 3 because it might have a q value based off one visit, which is a very poor estimate so it makes it more likely to resign incorrectly.

Reply:
mm, I see what you mean.